Assets/Scripts/NPCs/NPCManager.cs

```diff
@@ -7,29 +10,61 @@ public class NPCManager : MonoBehaviour
     private Rigidbody _rigidbody;
     private Animator _animator;
     private AudioSource _audioSource;
+    private NPCStateMachine _stateMachine;
+    
+    private int _punchLayerIndex;
+
+    private bool _canChase = true;
+    private bool _canPunch;
     
     //For some reason, some models are faster than others
     [SerializeField] private float ModelSpeedMultiplier;
-
+    [SerializeField] private AudioClip[] bumpingSounds;
+    
     private bool _lookingForPlayer = false;
+    
+    private bool _punching;
 
     public Rigidbody PlayerRigidbody => _playerRigidbody;
     public Rigidbody Rigidbody => _rigidbody;
     public Animator Animator => _animator;
     public AudioSource AudioSource => _audioSource;
-
+    public int PunchLayerIndex => _punchLayerIndex;
     public bool LookingForPlayer
     {
         get => _lookingForPlayer;
         set => _lookingForPlayer = value;
     }
 
+    public bool Punching
+    {
+        get => _punching;
+        set => _punching = value;
+    }
     void Start()
     {
         _playerRigidbody = Indestructibles.Player.GetComponent<Rigidbody>();
         _rigidbody = GetComponent<Rigidbody>();
         _animator = GetComponent<Animator>();
         _audioSource = GetComponent<AudioSource>();
+        _stateMachine = GetComponent<NPCStateMachine>();
+        _punchLayerIndex = _animator.GetLayerIndex("Punch Layer");
+    }
+
+    private void Update()
+    {
+        var stateInfo = Animator.GetCurrentAnimatorStateInfo(PunchLayerIndex);
+        
+        // To let the punching animation finish 
+        if (!Punching && stateInfo.normalizedTime % 1 < 0.1f)
+        {
+            Animator.SetLayerWeight(PunchLayerIndex,0.0f);
+        }
+
+        if (Indestructibles.PlayerData.IsKnockedOut)
+        {
+            _stateMachine.StartWandering();
+        }
```

Comment:
Do we even need to change the state of the NPCs? The game ends when the player is knocked out

Reply:
This is just for better visual look, this way the NPC won't push on the dead player :)